sdk/search/Azure.Search.Documents/src/AzureSearchDocumentsEventSource.cs

```diff
@@ -15,35 +18,94 @@ internal sealed class AzureSearchDocumentsEventSource : EventSource
         {
-            WriteEvent(PendingQueueResizedEvent, queueSize);
+            WriteEvent(PendingQueueResizedEvent, publisherType, queueSize);
         }
 
-        [Event(RetryQueueResizedEvent, Level = EventLevel.Verbose, Message = "Indexing publisher retry queue size = {0}.")]
-        public void RetryQueueResized(int queueSize)
+        [Event(RetryQueueResizedEvent, Level = EventLevel.Verbose, Message = "{0}: retry queue size = {1}.")]
+        public void RetryQueueResized(string publisherType, int queueSize)
         {
-            WriteEvent(RetryQueueResizedEvent, queueSize);
+            WriteEvent(RetryQueueResizedEvent, publisherType, queueSize);
         }
 
-        [Event(BatchSubmittedEvent, Level = EventLevel.Informational, Message = "Indexing publisher at {0} has submitted batch of size {1}.")]
-        public void BatchSubmitted(string endPoint, int batchSize)
+        [Event(BatchSubmittedEvent, Level = EventLevel.Informational, Message = "{0} at {1} has submitted batch of size {2}.")]
+        public void BatchSubmitted(string publisherType, string endPoint, int batchSize)
         {
-            WriteEvent(BatchSubmittedEvent, endPoint, batchSize);
+            WriteEvent(BatchSubmittedEvent, publisherType, endPoint, batchSize);
         }
 
-        [Event(BatchActionCountUpdatedEvent, Level = EventLevel.Informational, Message = "Indexing publisher at {0} has updated the starting batch action count from {1} to {2}.")]
-        public void BatchActionCountUpdated(string endPoint, int oldBatchCount, int newBatchCount)
+        [Event(BatchActionCountUpdatedEvent, Level = EventLevel.Warning, Message = "{0} at {1} has updated the starting batch action count from {2} to {3}.")]
+        public void BatchActionCountUpdated(string publisherType, string endPoint, int oldBatchCount, int newBatchCount)
         {
-            WriteEvent(BatchActionCountUpdatedEvent, endPoint, oldBatchCount, newBatchCount);
+            WriteEvent(BatchActionCountUpdatedEvent, publisherType, endPoint, oldBatchCount, newBatchCount);
+        }
+
+        [Event(PublisherDisposedWithPendingActionsEvent, Level = EventLevel.Error, Message = "{0}: {1} unsent indexing actions at {2}.")]
+        public void PublisherDisposedWithPendingActions(string componentType, string endPoint, int indexingActionsCount)
+        {
+            WriteEvent(PublisherDisposedWithPendingActionsEvent, componentType, indexingActionsCount, endPoint);
+        }
+
+        [Event(DocumentsPublishedEvent, Level = EventLevel.Verbose, Message = "{0}: publishing documents. Flush = {2}")]
+        public void PublishingDocuments(string publisherType, bool flush)
+        {
+            WriteEvent(DocumentsPublishedEvent, publisherType, flush);
+        }
+
+        [Event(ActionNotificationEventHandlerExceptionThrownEvent, Level = EventLevel.Error, Message = "{0}: exception thrown at {1}. Action = {2}. Exception: {3}")]
```

Comment:
Why a `:` for Exception when `=` for Action?  And is this saying the exception is thrown at the endpoint?  That reads confusingly to me.

Reply:
I have changed it to say 'for' instead of 'at'. Please see if that reads better.

Reply:
That sounds good.